helpers/voting/resolveAncillaryData.ts

```diff
@@ -0,0 +1,183 @@
+    ...request,
+    ancillaryData: resolvedAncillaryData[i],
+  }));
+}
+
+export async function resolveAncillaryData(
+  args: Pick<RequestAddedEvent["args"], "ancillaryData" | "time" | "identifier">
+): Promise<string> {
+  const decodedAncillaryData = decodeHexString(args.ancillaryData);
+  const { ancillaryDataHash, childOracle, childChainId, childBlockNumber } =
+    extractMaybeAncillaryDataFields(decodedAncillaryData);
+
+  if (ancillaryDataHash && childOracle && childChainId && childBlockNumber) {
+    try {
+      // if decoded ancillary data contains these fields, then we must extract from spoke
+      const _childOracle = `0x${childOracle}`;
+      const _childChainId = Number(childChainId);
+      const _childBlockNumber = Number(childBlockNumber);
+
+      const parentRequestId = keccak256(
+        defaultAbiCoder.encode(
+          ["bytes32", "uint256", "bytes"],
+          [args.identifier, args.time, args.ancillaryData]
+        )
+      );
+
+      return fetchAncillaryDataFromSpoke({
+        parentRequestId,
+        childOracle: _childOracle,
+        childChainId: _childChainId,
+        childBlockNumber: _childBlockNumber,
+      });
+    } catch (error) {
+      console.error("Unable to resolve original ancillary data", {
+        at: "resolveAncillaryData()",
+        data: args,
+        cause: error,
+      });
+
+      return args.ancillaryData;
+    }
+  }
+
+  return args.ancillaryData;
+}
+
+const AncillaryDataFieldsTypes = {
+  // strings
+  ancillaryDataHash: "w",
+  childOracle: "w",
+  childRequester: "w",
+  // numbers
+  childBlockNumber: "d",
+  childChainId: "d",
+} as const;
+
+type AncillaryDataField = keyof typeof AncillaryDataFieldsTypes;
+
+export function getAncillaryDataField(
+  decodedAncillaryData: string,
+  field: AncillaryDataField
+): string | undefined {
+  const pattern = new RegExp(
+    `${field}:(\\${AncillaryDataFieldsTypes[field]}+)`
+  );
+  const match = decodedAncillaryData.match(pattern) ?? [];
+  return match?.[1];
+}
+
+function extractMaybeAncillaryDataFields(decodedAncillaryData: string) {
+  return Object.fromEntries(
+    Object.keys(AncillaryDataFieldsTypes).map((key) => [
+      key as AncillaryDataField,
+      getAncillaryDataField(decodedAncillaryData, key as AncillaryDataField),
+    ])
+  );
+}
+
+async function fetchAncillaryDataFromSpoke(args: {
+  parentRequestId: BytesLike;
+  childOracle: string;
+  childChainId: number;
+  childBlockNumber: number;
+}): Promise<string> {
+  const provider = getProvider(args.childChainId);
+  // TODO: install from upgraded @uma/contracts-frontend pkg
+  const OracleSpoke = new Contract(
+    args.childOracle,
+    new Interface(abi),
+    provider
+  );
+  const filter = OracleSpoke.filters.PriceRequestBridged(
+    null,
+    null,
+    null,
+    null,
+    null,
+    args.parentRequestId
+  );
+
+  const events = await OracleSpoke.queryFilter(
+    filter,
+    args.childBlockNumber - 1,
```

Comment:
its not strictly required to query from previous block, it can be the same as to block

Reply:
Interesting, I did try this and it fails to find anything

Reply:
@Reinis-FRP https://github.com/UMAprotocol/voter-dapp-v2/pull/364/commits/65c2211795b3bb6bbbf045c73454c0ac8bf2700e

Reply:
this should do it!

Reply:
thats weird, cause they to and from should be inclusive.